apollo-router/src/plugins/reporting.rs

```diff
@@ -1,30 +1,153 @@
-use crate::apollo_telemetry::new_pipeline;
-#[cfg(any(feature = "otlp-grpc", feature = "otlp-http"))]
-use crate::configuration::otlp;
 #[cfg(any(feature = "otlp-grpc", feature = "otlp-http"))]
-use crate::configuration::otlp::Tracing;
-use crate::configuration::JaegerEndpoint;
-use crate::configuration::OpenTelemetry;
-use crate::configuration::SpaceportConfig;
-use crate::configuration::StudioGraph;
+mod otlp;
+
+use crate::apollo_telemetry::new_pipeline;
+use crate::apollo_telemetry::SpaceportConfig;
+use crate::apollo_telemetry::StudioGraph;
+use crate::configuration::{default_service_name, default_service_namespace};
 use crate::set_subscriber;
 use crate::GLOBAL_ENV_FILTER;
 use apollo_router_core::{register_plugin, Plugin};
 use apollo_spaceport::server::ReportSpaceport;
+use derivative::Derivative;
 use futures::Future;
-use opentelemetry::sdk::trace::BatchSpanProcessor;
+use opentelemetry::sdk::trace::{BatchSpanProcessor, Sampler};
+use opentelemetry::sdk::Resource;
 use opentelemetry::trace::TracerProvider;
+use opentelemetry::KeyValue;
+/*
+#[cfg(any(feature = "otlp-grpc", feature = "otlp-http"))]
+use otlp;
+*/
```

Comment:
to delete

Reply:
Hmm, not sure what happened here, but: [e705cb6](https://github.com/apollographql/router/pull/541/commits/e705cb61460d73b16e3e97026fd0b13f45f7aebb)